koku/api/resource_types/azure_regions/view.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright 2021 Red Hat Inc.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""View for Azure Region locations."""
+from django.db.models import F
+from django.utils.decorators import method_decorator
+from django.views.decorators.vary import vary_on_headers
+from rest_framework import filters
+from rest_framework import generics
+
+from api.common import CACHE_RH_IDENTITY_HEADER
+from api.common.permissions.resource_type_access import ResourceTypeAccessPermission
+from api.resource_types.serializers import ResourceTypeSerializer
+from reporting.provider.azure.models import AzureCostSummaryByLocation
+
+
+class AzureRegionView(generics.ListAPIView):
+    """API GET list view for Azure Region locations."""
+
+    queryset = (
+        AzureCostSummaryByLocation.objects.annotate(**{"value": F("resource_location")}).values("value").distinct()
```

Comment:
We probably need to handle these the same way we did the aws one and filter out null values. This is how we filtered out null values for aws region:
`.filter(region__isnull=False)`